build/scripts/PullRequestValidation/ValidateIssuesForPullRequest.ps1

```diff
@@ -0,0 +1,47 @@
+using module ..\GitHub\GitHubPullRequest.class.psm1
+using module ..\GitHub\GitHubIssue.class.psm1
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string] $PullRequestNumber,
+    [Parameter(Mandatory = $true)]
+    [string] $Repository
+    )
+
+# Set error action
+$ErrorActionPreference = "Stop"
+
+Write-Host "Validating PR $PullRequestNumber"
+
+$pullRequest = [GitHubPullRequest]::Get($PullRequestNumber, $Repository)
+$prDescription = $pullRequest.GetBody()
+$issuesStartingPoint = "Fixes #"
+
+$issueStartingIndex = -1
+if($prDescription) {
+    $issueStartingIndex = $prDescription.IndexOf($issuesStartingPoint)
+}
+
+if ($issueStartingIndex -eq -1) {
+    throw "Could not find issues section in the pull request description. Please make sure the pull request description contains a line that contains 'Fixes #' followed by the issue number being fixed"
+}
+
+$issuesDescription = $prDescription.Substring($issueStartingIndex + $($issuesStartingPoint.Length) - 1)
+$issueIds = $issuesDescription.Split(' ,') | Where-Object { $_ -match "#\d+" } | ForEach-Object { [int] $_.Trim("# ") }
+
+foreach ($issueId in $issueIds) {
+    Write-Host "Validating issue $issueId"
+    $issue = [GitHubIssue]::Get($issueId, $Repository)
+
+    $Comment = "Issue $($issue.html_url) is not approved. Please make sure the issue is approved before continuing with the pull request."
+    if (-not $issue.IsApproved()) {
+        $pullRequest.AddComment($Comment)
+
+        throw "$Comment"
```

Comment:
Bit of an edge case.. But would it make sense to let the foreach loop keep running and throw at the end? That way we can create a comment for all non-approved issues in case there are multiple

Reply:
Good one, will do.